### PR TITLE
(test) O3-2297: Add tests for `visit-detail.component.tsx`

### PR DESCRIPTION
**Pull request by ayush-AI**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

In this PR I have added tests for `visit-detail.component.tsx`

## Screenshots

*None.*

## Related Issue

[O3-2297](https://issues.openmrs.org/browse/O3-2297)

## Other

*None.*

### CODE REVIEW

**Comment by ayush-AI** [comment]
cc/ @Piumal1999 @anjula-sack @jayasanka-sack

File: packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import { render, screen, act } from '@testing-library/react';
+import VisitDetailComponent from './visit-detail.component';
+
+jest.mock('./visit.resource', () => ({
+  useVisit: jest.fn(),
+}));
+
+describe('VisitDetailComponent', () => {
+  const visitUuid = 'some-visit-uuid';
+  const patientUuid = 'some-patient-uuid';
```
Comment (Piumal1999) [inline packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx:11]: For these kinds of variables, let's use `uuid`s similar to actual `uuid`s. (I think there's a specific format and the number of characters in openmrs uuids)

File: packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  it('renders loading indicator when data is loading', () => {
+    require('./visit.resource').useVisit.mockReturnValue({
+      visit: null,
+      isLoading: true,
+    });
```
Comment (Piumal1999) [inline packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx:17]: We can mock it like this:

```
useVisit.mockReturnValue({
      visit: null,
      isLoading: true,
});
```

File: packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        encounters: [],
+      },
+      isLoading: false,
+    });
+
+    render(<VisitDetailComponent visitUuid={visitUuid} patientUuid={patientUuid} />);
+
+    expect(screen.getByText(/Some Visit Type/)).toBeInTheDocument();
+    expect(screen.getByText(/29-Jul-2023, 12:34 PM/)).toBeInTheDocument();
+    screen.debug();
```
Comment (Piumal1999) [inline packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx:39]: Can you elaborate on the usage of this line?

File: packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx (hunk 2 above)
Comment (Piumal1999) [reply packages/esm-active-visits-app/src/visits-summary/visit-detail.test.tsx:17]: Let's do the same to other `mockReturnValue` function calls as well

**Review by Piumal1999 — CHANGES_REQUESTED** [review]

**Comment by ayush-AI** [comment]
@Piumal1999 I updated the PR please have a look.

**Review by Piumal1999 — APPROVED** [review]
Thank you @ayush-AI

**Comment by jayasanka-sack** [comment]
@Piumal1999 @ayush-AI Tests have been failing since this PR. Could you please fix it as soon as possible?

**Pull request merged**